From: Esben Haabendal <esben@haabendal.dk>
Date: Fri, 7 Nov 2014 14:15:03 +0100

Fix NOHZ local_softirq_pending 08 warning

This patch replaces netif_rx() with netif_rx_ni() which has to be used
from process/softirq context.

See Linux kernel commit 481a8199142c for similar fix.  According to it,
this might fix the "NOHZ local_softirq_pending 08" warnings that has been
observed.

Signed-off-by: Esben Haabendal <esben@haabendal.dk>

diff --git a/master/debug.c b/master/debug.c
--- a/master/debug.c
+++ b/master/debug.c
@@ -196,7 +196,7 @@ void ec_debug_send(
     skb->dev = dbg->dev;
     skb->protocol = eth_type_trans(skb, dbg->dev);
     skb->ip_summed = CHECKSUM_UNNECESSARY;
-    netif_rx(skb);
+    netif_rx_ni(skb);
 }
 
 /******************************************************************************
diff --git a/master/ethernet.c b/master/ethernet.c
--- a/master/ethernet.c
+++ b/master/ethernet.c
@@ -688,7 +688,7 @@ void ec_eoe_state_rx_fetch(ec_eoe_t *eoe
         eoe->rx_skb->dev = eoe->dev;
         eoe->rx_skb->protocol = eth_type_trans(eoe->rx_skb, eoe->dev);
         eoe->rx_skb->ip_summed = CHECKSUM_UNNECESSARY;
-        if (netif_rx(eoe->rx_skb)) {
+        if (netif_rx_ni(eoe->rx_skb)) {
             EC_SLAVE_WARN(eoe->slave, "EoE RX netif_rx failed.\n");
         }
         eoe->rx_skb = NULL;
